src/Plugins/Loader/ManagedLoadContext.cs

```diff
@@ -309,7 +317,62 @@ private bool SearchForLibrary(NativeLibrary library, string prefix, out string?
         private IntPtr LoadUnmanagedDllFromResolvedPath(string unmanagedDllPath)
         {
             var normalized = Path.GetFullPath(unmanagedDllPath);
-            return LoadUnmanagedDllFromPath(normalized);
+
+            // Create shadow copy directory only when hot reloading is enabled and plugins are
+            // dependent on unmanaged DLL(s).
+            if (_loadInMemory && string.IsNullOrEmpty(_unmanagedDllShadowCopyDirectoryPath))
+            {
+                _unmanagedDllShadowCopyDirectoryPath = CreateTempDirectory();
+            }
+
+            return _loadInMemory
+                ? LoadUnmanagedDllFromShadowCopy(normalized)
+                : LoadUnmanagedDllFromPath(normalized);
+        }
+
+        private IntPtr LoadUnmanagedDllFromShadowCopy(string unmanagedDllPath)
+        {
+            var shadowCopyDllPath = CreateShadowCopy(unmanagedDllPath);
+
+            return LoadUnmanagedDllFromPath(shadowCopyDllPath);
+        }
+
+        private string CreateShadowCopy(string dllPath)
+        {
+            var dllFileName = Path.GetFileName(dllPath);
+            var shadowCopyPath = Path.Combine(_unmanagedDllShadowCopyDirectoryPath, dllFileName);
+
+            File.Copy(dllPath, shadowCopyPath);
+
+            return shadowCopyPath;
+        }
+
+        private string CreateTempDirectory()
+        {
+            var tempDirectoryName = Path.GetTempFileName();
+            File.Delete(tempDirectoryName);
+
+            var tempDirectory = Directory.CreateDirectory(tempDirectoryName);
+
+            return tempDirectory.FullName;
+        }
+
+        private void OnUnloaded()
+        {
+            if (string.IsNullOrEmpty(_unmanagedDllShadowCopyDirectoryPath))
```

Comment:
If you take my other suggestions, this would probably need to be `if (_loadInMemory && Directory.Exists(_unmanagedDllShadowCopyDirectoryPath))`